Mathlib/Data/Nat/Choose/Sum.lean

```diff
@@ -74,6 +75,32 @@ theorem add_pow [CommSemiring R] (x y : R) (n : ℕ) :
     (x + y) ^ n = ∑ m ∈ range (n + 1), x ^ m * y ^ (n - m) * n.choose m :=
   (Commute.all x y).add_pow n
 
+/-- A special case of the **binomial theorem** -/
+theorem sub_pow [CommRing R] (x : R) (y : R) (n : ℕ) :
+    (x - y) ^ n = ∑ m ∈ range (n + 1), (-1) ^ (m + n) * n.choose m • x ^ m * y ^ (n - m) := by
+  rw [sub_eq_add_neg, add_pow]
+  apply Finset.sum_congr rfl
+  intro m hm
+  simp only [mem_range] at hm
+  rw [neg_pow]
+  ring_nf
+  nth_rewrite 3 [mul_assoc]
+  congr
+  rw [← mul_one (a := (-1) ^ (n - m))]
+  nth_rewrite 2 [show (1 : R) = (-1) ^ (2 * m) by simp]
+  rw [← pow_add, ← pow_add, show n - m + 2 * m = m + n by omega]
+
+/-- A special case of the sub_pow -/
+theorem sub_square_pow [CommRing R] (x : R) (n : ℕ) :
+    (x - x ^ 2) ^ n = ∑ m ∈ range (n + 1), (- 1) ^ m * n.choose m • x ^ (n + m) := by
```

Comment:
I don't really see the value in this lemma. You get this just by applying `sub_pow` and rewriting the terms in the obvious fashion. Can you inline it in your application?

Reply:
emmmmm. I get it! It must be ok!

Reply:
I worried that my proof is too long, so I separate it. ok. I'll inline it in my lemma.